shard.lua

```diff
@@ -2098,6 +2119,46 @@ local function init(cfg, callback)
     return true
 end
 
+local function wait_connection()
+    pool:wait_connection()
+    on_connected(configuration)
+    if configuration.monitor == true then
+        pool:start_monitoring()
+    end
+    connection_fiber.state = 'connected'
+    return true
+end
+
+local function wait_connected(timeout, delay)
```

Comment:
What is difference between wait_connection and wait_connected? Can you please add comments? An original shard was very hard to read and update the mostly because of such functions, which do something without any comments.